File: 415/solution.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        i = len(num1) - 1
        j = len(num2) - 1
        add = 0
        res = []
        while i >= 0 or j >= 0:
            val1 = int(num1[i]) if i >= 0 else 0
            val2 = int(num2[j]) if j >= 0 else 0
            total = add + val1 + val2
            res.append(str(total % 10))
            add = total // 10
            i -= 1
            j -= 1
        if add > 0:
            res.append(str(add))
        res.reverse()
        return "".join(res)

File: 415/Solution.java
class Solution {
    public String addStrings(String num1, String num2) {
        int i = num1.length() - 1;
        int j = num2.length() - 1;
        StringBuffer ans = new StringBuffer();
        int add = 0;
        while (i >= 0 || j >= 0) {
            int val1 = i >= 0 ? num1.charAt(i) - '0': 0;
            int val2 = j >= 0 ? num2.charAt(j) - '0': 0;
            int total = add + val1 + val2;
            ans.append(total % 10);
            add = total / 10;
            i--;
            j--;
        }
        if (add > 0) {
            ans.append(add);
        }
        ans.reverse();
        return ans.toString();
    }
}